### DISCUSSION TITLE
Welcome to TVerRec Discussions!

### DISCUSSION THREAD

**Discussion by dongaba** [dongaba/TVerRec#1] [General]
## 👋 ようこそ!
  ディスカッションを、コミュニティの他のメンバーとつながるための場として活用しています。ご自由にどうぞ。
  * 疑問に思っていることを質問する
  * アイデアを投稿する
  * 他のコミュニティメンバーと交流する

## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

**Comment by pkqah** [comment]
dongabaさん、こんにちは！
お伺いしたいことがあるのですが、なぜwin/start_tverrec.batを実行すると’設定ファイルの読み込みに失敗しました’ということが表示されますか。

**Reply by dongaba** [reply]
どうやらWindows PowerShellではちゃんと動かないバグがありそうです。
プログラムの方は近日中に修正しますが、お急ぎの場合PowerShell Coreをインストールしてから再度実行してみてもらえますか。
https://docs.microsoft.com/ja-jp/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.2

**Reply by dongaba** [reply]
v2.3.1 で修正しました。不具合が残っていたらお知らせください。

**Reply by pkqah** [reply]
dongabaさん、v2.3.1はもう大丈夫で、ダウンロードできます。
お疲れ様でした。ありがとうございました！

**Comment by nmt3325** [comment]
よろしくお願いします windows11 ver2.3.9使用中です
ツールが自動ダウンロードされませんでした。
Convert-Path : Cannot find path 'C:\Users\nmtki\Downloads\TVerRec-2.3.9\bin\youtube-dl.exe' because it does not exist.
と表示されます。同様のエラーがffmpegでも表示されたため、ダウンロードしてTVerRec-2.3.9\binフォルダに配置しましたが、youtbe-dl.exeについては「必要なツールは自動的にダウンロードされますが、うまくいかない場合は以下から取得してください。」のリンクにあった
youtube-dl-pathed [https://github.com/ytdl-patched/ytdl-patched/releases]　のリンクのどのファイルをyoutbe-dl.exeをリネームして配置すべきでしょうか？

**Reply by dongaba** [reply]
youtube-dlが正常にダウンロードできない問題は認識しており、本日中めどで更新版をリリースする予定です。
それまでの間は、以下をダウンロードして
[ytdl-patched-red.exe](https://github.com/ytdl-patched/ytdl-patched/releases/download/1650456893/ytdl-patched-red.exe)を「youtube-dl.exe」にリネームしてお使いください。アイコンしか違わないのでytdl-patched-white.exeを使っていただいてもかまいません。
ffmpegについて問題を認識していなかったので確認の上修正します。

**Reply by dongaba** [reply]
v2.4.0でyoutube-dlとffmpegがダウンロードされない問題は修正できたと思います。ご確認ください。

**Reply by nmt3325** [reply]
設定ファイルの読み込みyoutube-dl.exeのダウンロードともに改善されました。
ありがとうございます。

**Comment by nmt3325** [comment]


**Reply by dongaba** [reply]
いろいろと不具合があったようですみません。
v2.4.0で修正できたと思うので、v2.4.0をダウンロードして再度実行してみていただけますか。

**Reply by nmt3325** [reply]
ありがとうございます
すみませんあまりにもエラーが長く見づらそうなので削除させていただきました

**Reply by dongaba** [reply]
了解です、ご配慮ありがとうございます。

**Comment by nmt3325** [comment]
前回の修正ありがとうございます。
個人的な要望なのですが、ダウンロード先にチャンネルごとにフォルダ分けした上で番組フォルダを作るオプションをつけるとかなり見やすくなると思います。
例　TBS/水曜日のダウンタウン/水曜日のダウンタウン.mp4
のような感じです。
あとダウンロード後のチェック処理もスキップするオプションがあると嬉しいです。

**Reply by dongaba** [reply]
コメントありがとうございます。
チャンネルごとのフォルダ分けはそれほど難しくないので検討します。
ダウンロードとのチェック処理は時間を短縮する案があるにはあるので、こちらも検討します。
ただ、動画ファイルが壊れていることもわりと多く、処理速度を早くすることととのトレードになりますね。
時間がかかるのが嫌、というばあいはデコードオプションを指定してハードウェアデコードをすると早くなるかもしれません。
例えば、ここ数年のIntelのCPUを使ったWindows PCであれば、user_setting.ps1で「$script:ffmpegDecodeOption = '-hwaccel qsv -c:v h264_qsv'」を指定すると、CPU使用率を下げつつ処理時間が短縮するかもしれないです。

**Reply by dongaba** [reply]
ご要望の機能をv2.4.2に追加しました。こんな感じでイメージ通りですかね?

**Reply by nmt3325** [reply]
はい、イメージと全く同じです。ありがとうございます。
これからのご活躍祈念しております。

**Comment by BRABUS-RC** [comment]
dongaba様。
徘徊中にこちらに伺いソフト頂きました。　設定で自動保存凄いですね。
番組IDで設定したいと思いますが、URLに「series/srxxxxxxxx」が無いのですが・・・
こちらならあります。「episodes/epxxxxxxxx」
番組ID見る方法ありましたらお願い致します。

**Reply by nmt3325** [reply]
それはたぶん番組自体のurlではないかと思います。右上に番組トップへというボタンが表示されていると思います。
そこを押すと、https://tver.jp/series/srdet3e55w　このようなurlが表示されます。

**Reply by BRABUS-RC** [reply]
dongaba様。
お忙しい中ありがとうございました。
快調に稼働しました。
今後ともよろしくお願い致します。

**Reply by nmt3325** [reply]
dongabaさんに感謝感謝

**Reply by BRABUS-RC** [reply]
NKMTIC様　
教えて頂いたのにすみません。ありがとうございました。
dongabaさんに感謝すると共にNKMTICさん宜しくお願い致します。

**Reply by BRABUS-RC** [reply]
すみません。本日ソフトを起動しましたら全てDL出来なくなってました。何か考えられる原因ありますでしょか。

**Reply by dongaba** [reply]
youtube-dlの最新版が壊れているようです。
2日ほど前にリリースされたyoutube-dlは動くようなので、[[https://github.com/ytdl-patched/ytdl-patched/releases/tag/1650811782](https://github.com/ytdl-patched/ytdl-patched/releases/tag/1650811782)]から取得して置き換える、ということもできますが、自動Update機能で不具合のある最新版に置き換えられてしまうのでyoutube-dl側で不具合を修正するのを待つしかないです。
数日待ってもyoutube-dl側で対応されないようであれば、自動Update機能をOffにするオプションを追加します。

**Reply by BRABUS-RC** [reply]
dongabaさん、不具合なんですね。DL出来なくなったのであせりました。
youtube-dl側の対応が出来ない時は宜しくお願い致します。
ありがとうございました。

**Reply by BRABUS-RC** [reply]
連投すみません。動画を1本ずつダウンロードするソフトで、再ダウンロードが出来る様になりませんか。
昨日欲しい物を一本ダウンロードしましたが、youtube-dlの不具合でダウンロード出来ませんでした。
youtube-dlが改善されて欲しい物がダウンロード出来ない（過去に処理したビデオです。スキップしますとなる）ので再ダウンロードが出来る様になったらうれしいです。
すみませんが宜しくお願い致します。

**Reply by dongaba** [reply]
READMEの記載が不十分で申し訳ありません。
win/b.validate.batを実行すればダウンロードできていないファイルがある場合(正確にはダウンロードしたビデオファイルが破損している場合)に、ダウンロード済みリストからダウンロード履歴を削除するので、再度ダウンロードできるようになります。
お試しください。

**Reply by BRABUS-RC** [reply]
dongaba様　試してみます。
今Recを2.4.3に更新しましたらWindowsセキュリティに反応する様になりました・・・
赤文字で下記が表示します。
checkRequiredFile : youtube-dlが存在しません。終了します。
At D:\TVerRec-master\TVerRec-master\src\tverrec_single_5.ps1:104 char:1
+ checkRequiredFile            #設定で指定したファイル・フォルダの存在チェック
+ ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,checkRequiredFile

宜しくお願い致します。

**Reply by dongaba** [reply]
私の環境では再現しないですね。。
binディレクトリ配下にyoutube-dl.exeは存在しているでしょうか?

**Reply by BRABUS-RC** [reply]
Windowsセキュリティ　リアルタイム保護をオンのままソフトを起動すると（１本づつDLソフト）下記メッセージが表示しオフでソフトを起動すると正常に動きます。


Program 'youtube-dl.exe' failed to run: Operation did not complete successfully because the file contains a virus or po
tentially unwanted softwareAt D:\TVerRec-master\TVerRec-master\src\functions\update_ytdl-patched_5.ps1:40 char:31
+     $local:ytdlCurrentVersion = (& $local:ytdlPath --version)
+                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~.
At D:\TVerRec-master\TVerRec-master\src\functions\update_ytdl-patched_5.ps1:40 char:2
+     $local:ytdlCurrentVersion = (& $local:ytdlPath --version)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException
    + FullyQualifiedErrorId : NativeCommandFailed

The variable '$local:ytdlCurrentVersion' cannot be retrieved because it has not been set.
At D:\TVerRec-master\TVerRec-master\src\functions\update_ytdl-patched_5.ps1:50 char:40
+ Write-ColorOutput "youtube-dl current: $local:ytdlCurrentVersion"
+                                        ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (local:ytdlCurrentVersion:String) [], RuntimeException
    + FullyQualifiedErrorId : VariableIsUndefined

youtube-dl latest: 2022.04.27.1651026004
The variable '$local:ytdlCurrentVersion' cannot be retrieved because it has not been set.
At D:\TVerRec-master\TVerRec-master\src\functions\update_ytdl-patched_5.ps1:54 char:30
+ if ($local:latestVersion -eq $local:ytdlCurrentVersion) {
+                              ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (local:ytdlCurrentVersion:String) [], RuntimeException
    + FullyQualifiedErrorId : VariableIsUndefined

ffmpeg current: 5.0.1
ffmpeg latest: 5.0.1
ffmpegは最新です。

ビデオURLを入力してください。:

ソフトの確認お願いできませんか。

**Reply by BRABUS-RC** [reply]
youtube-dl.exeはあります。

**Reply by dongaba** [reply]
youtube-dl.exeがWindowsセキュリティによってウイルス判定されてしまっているようですので、これは私のTVerRecの問題ではないですね。
私のWindows11環境では同様の問題は起きていないので原因はわかりませんが、D:\TVerRec-master\TVerRec-master\bin\youtube-dl.exeを例外登録してWindowsセキュリティにブロックされないようにすると良いかもしれませんね。

**Reply by BRABUS-RC** [reply]
今また見たらyoutube-dl.exeが消えてます・・・
↑の設定試してみます。

**Reply by BRABUS-RC** [reply]
dongaba様。
解決しました。　お手数おかけしました。
ありがとうございました。

**Comment by nmt3325** [comment]
このプログラム自体を最新バージョンに更新しようと思った場合、ダウンロード先やダウンロード済の動画の情報などを保持したままにするにはどのようにすればよいでしょうか？

**Reply by dongaba** [reply]
ダウンロード先の情報はconf/user_setting.ps1にあるので、このファイルを上書きしなければダウンロード先は保持されます。
ダウンロード済の動画の情報はdb/tver.csvにあるので、こちらも上書きしなければ保持されます。

**Comment by BRABUS-RC** [comment]
dongaba様。お世話になります。
個別ダウンロード版を使い（Normal設定にしてダウンロードの進捗確認しながら）ダウンロードしてますが、昨日からダウンロード出来ません。dongabaさんは問題ありませんでしょか。
個別ダウンロード版にURLを貼り付け後、別窓が表示され進捗が表示されますが進捗の表示が消えたらダウンロード完了と考えてよろしいのですか。

**Reply by dongaba** [reply]
私の環境では引き続きダウンロードできています。(とは言え、私は個別ダウンロード版は使ってなくて、一括ダウンロードを使っていますが)
さて、ご質問の件ですが、別窓が消えたらダウンロード完了とは言えない場合があります。
ダウンロード中に何らかの問題(例えばTVer側との接続を切られた場合など)には、別窓は消えますがダウンロードは完了していない状態となります。
そのような中途半端な状態でダウンロードが中断されてしまった場合、このツール自体はダウンロードが終わったものと思って同じ動画のダウンロードをスキップするようになっていますので、同じ動画の再ダウンロードがされなくなってしまいます。
そのような状態になった際にはwin/c.validate.batを実行すればダウンロードできていないファイルがある場合(正確にはダウンロードしたビデオファイルが破損している場合)に、ダウンロード済みリストからダウンロード履歴を削除するので、再度ダウンロードできるようになります。
お試しください。
また、ダウンロードが中断した際のゴミファイルを掃除するにはwin/b.delete_video.batを実行してください。

**Reply by BRABUS-RC** [reply]
dongabaさん　こんばんは。
アドバイスありがとうございました。
アドバイスに従い再ダウンロードしてますが、一括、個別もダウンロード出来なくなりました。
今もダウンロードを試しましたが下記表示のままソフトが何も反応して無いです。
「ffmpegをダウンロードします。 https://www.gyan.dev/ffmpeg/builds/ffmpeg-release-essentials.zip」

調子が悪いのでHPよりソフトを再ダウンロードし「conf」の中身をダウンロード出来てた時の物をコピーして作業をしてます。

**Reply by BRABUS-RC** [reply]
すみません。
無事に↑の「.zip」ダウンロード出来動画のダウンロードが出来る様になりました。
お騒がせしました。

**Comment by tesz1** [comment]
いつもありがたく使わせていただいております。ありがとうございます。
windows10、ver2.4.8でジャンルでアニメ1個のみ指定して使用しているのですが特定の動画(ちびまる子ちゃん、デジモンゴーストゲームなど)で
exit code: 1    error count: 1
が出て失敗しているようなのですが何が原因でしょうか？
5月30日に実行したときには成功していました。よろしくお願いします。

**Reply by dongaba** [reply]
exit code: 1 error count: 1 というのは、動画ダウンロード後の動画ファイルの検証時にエラーが出ている状態です。
デジモンゴーストゲームは録画していないのですが、ちびまる子ちゃんは私も録画していますので確認しましたが、
直近では5/22、5/29、6/5の放送分がちゃんと録画(ダウンロード)されていますので、あなたの環境に依存しているような気がします。

error count: 1 というのは、一般的には動画ファイルがダウンロードに失敗している場合に起きます。(ダウンロードしたファイルが破損しているのではなく、そもそもファイルが存在しない、ということです。)
動画ファイルができていない場合と動画ファイルが破損していて再生できない場合では被疑箇所が変わってきますのでuser_setting.ps1で $script:disableValidation = $true を設定し、動画の検証を行わずにダウンロードしてみていただけますか?

**Reply by tesz1** [reply]
さっそくの提案ありがとうございます。試してみたところ改善されませんでした。
そこで$script:windowShowStyle = 'Normal'に変えて試してみたところyoutube-dl.exeが起動して数秒時間を置いて高速でログが流れ
ERROR: [brightcove:new] Unable to download JSON metadata: HTTP Error 404: Not Found (caused by <HTTPError 404: 'Not Found'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
とエラーが出ている状態でした。
TVerRecでは無くyt-dlp側の問題ですかね・・・yt-dlpのVerを変えてみても変化ありませんでした。そして失敗してたのは全部フジテレビだったのですが関係ありますでしょうか？
最早言いがかりになってしまっていますがよろしければ知恵を貸していただけるとありがたいです。

**Reply by dongaba** [reply]
yt-dlpの問題は修正できませんが、yt-dlpの派生バージョンを使えるようにする変更をv2.4.9でリリースしますので、そちらを試してみてください。
$script:preferredYoutubedl に ytdl-patched を設定すると別バージョンのyoutube-dl (https://github.com/ytdl-patched/ytdl-patched) を使うようになります。こちらの方が頻繁に更新されており、不具合発生時やTVer等の仕様変更時にもすぐ対応される傾向があります。

あと、他にできそうな対策としては、同時ダウンロード数を少なくしてみるのも良いかもしれません。
もしかするとフジテレビ側からBANされている可能性もあるかもしれません。

**Reply by dongaba** [reply]
BANではなさそうですね。TVerのフジテレビ動画で仕様変更があったようです。
以下の掲示板の966、967あたりで言及されてます。
http://egg.5ch.net/test/read.cgi/streaming/1648663464/l50

**Reply by tesz1** [reply]
解決案提案から情報収集までありがとうございます。
自分でも情報を追って行きたいと思います、この度はご迷惑お掛けしました。何か進展がありましたらまた報告させていただきます。

**Reply by tesz1** [reply]
別バージョンのyoutube-dl (https://github.com/ytdl-patched/ytdl-patched)
の最新版にて無事成功しました。丁寧な助言ありがとうございました。

**Comment by ghost** [comment]
昨日はお世話になりました｡

先程試しに使ってみたのですが､　2.65→2.66,2.67に移行して使用してみたところ､リスト作成のx.generate_list.batを実行しても一瞬で終了してリストファイルが作成されませんでした｡
2.65以前の機能(一括DLなど)は引き続き新版でも利用できています｡

Verアップ時にconf(settingは書式が異なるため再作成),dbフォルダのみ引き継ぎしております｡
過去スレを参考にして移行しましたが､間違っているのでしょうか？

keywordには試しにnew/allを入れてみました｡番組名でも試しています｡


なお､もしよろしければTverRec自体のアップデートを自動で行うバッチの実装をお願いできますか？(バックアップZIP等作成付だとなおうれしいです)｡

お手数ですがよろしくお願いいたします｡

**Reply by dongaba** [reply]
dbフォルダ配下に新しいファイルが追加になっているので、dbフォルダをv2.6.7で上書きしてください。
ダウンロード履歴やユーザの設定(user_setting.ps1)は配布物に含まれていないので、バージョンアップの際は単純に上書きしてもらえればダウンロード履歴もユーザ設定も消えることはありません。
(とは言え、念のために事前にバックアップをとっていただくのが無難ではありますが)

**Reply by ghost** [reply]
先程試したのですが､引き継ぎ作業なしの新規セットアップでもx.generate_list.batが起動しませんでした｡
もし正常に動作する場合は他のバッチファイルのようにロゴが表示され､何らかの設定の不備がある場合にはエラーメッセージが返されるのでしょうか？
また､管理者権限で実行した場合､更に一瞬ログのようなものが見えますが､消えてしまいます｡
念のためですが､powershellcore7.31環境です｡

追記:ファイルはドライブ直下に親フォルダをおいています｡

**Reply by dongaba** [reply]
v2.6.7で追加した2つのBATファイルの改行コードがLFになってしまっていることが原因でした。
v2.6.8で改行コードの問題を修正したので、v2.6.8にバージョンアップして頂けますでしょうか。
ご迷惑をおかけし申し訳ありません。
ご報告をありがとうございました。

**Reply by ghost** [reply]
大変迅速な対応ありがとうございます｡

それと別件なのですが､a.download_video.batを実行したところ対象ファイルがDLされない上に完了フォルダ上の対象動画フォルダの中身が完全に消えてしまったのですが､バグでしょうか？
例　Tver/ネメシスのフォルダ内が全消失　
設定したTempフォルダ内には残骸なし　　
おまかんかもしれないので只今調査中です｡
追記
設定したTempフォルダ内には残骸なしでした｡
histry.csvには以前まで実行していたログと変わらず完了済になっています｡

**Reply by dongaba** [reply]
a.download_video.batは最近変更しておらず、私の環境でも動作しているのでおまかんの可能性が高いです。
また、a.download_video.batにはファイルを削除する機能はありませんので、ウイルス対策ソフト等の影響があるのかもしれません。
ファイルを削除する機能があるのはb.delete_trash.bat、c.validate_video.bat、d.move_video.batです。
(厳密にはa.download_video.batにもファイルを削除する処理はありますが、動画ファイルを削除するようなロジックはないです)
d.move_video.batを動作させたのであれば、設定ファイルの$script:saveBaseDirで設定したフォルダに移動された可能性はあります。
また、c.validate_video.batを動作させたのであれば、ダウンロードが不完全なファイルは削除されますが、その場合はhistory.csvからも当該番組のレコードが削除されます。

**Reply by ghost** [reply]
a.download_video.bat以外のバッチファイルは起動していません｡
ウイルス対策ソフトはデフォのDifender以外入れておりません｡
復元ソフト(recoverit)で確認したところ､新たにDLしようとした動画の一時ファイルもありませんでした｡
ただ､削除された動画は確認できました｡ゴミ箱にも入っていませんでした｡
また､手動での移動も行っていません｡

追記
また今朝の更新前より更新はしておらず､2.67環境で使用しておりました｡
再起動して実行したところ､同様にファイルが消える状況が確認できました｡

**Reply by dongaba** [reply]
githubよりv2.6.8をダウンロードし、基本的な設定だけを行った状態(つまり3種類のフォルダを指定しただけ)でa.download_video.batのみを実行して確認しましたが、ちゃんとダウンロードでき、ファイルも消えることがありませんでした。
設定ファイル、もしくは何らかのソフトや設定との相性が良くないのではないでしょうか。

**Reply by ghost** [reply]
最新版を落とし直して新規に設定をして動かしたら動きました｡
旧版と使うとまた消えたのですが､yt-dlが正常に動作していなかったのかもしれませんし､設定が悪かったのかもしれません｡
ありがとうございました｡

**Comment by ghost** [comment]
お世話になっております｡　何点か質問と希望なのですが

1. db/tver.csvおよびhistory.csvをデフォルトでBOM付きUTFにしていただくことは可能ですか？

2. listのデータ取得の件数を例えば"tag/variety"と定義した場合､HP上のさがす→バラエティ（tver.jp/tag**s**/variety）の検索結果の件数が違いますが､これはTver上の仕様によるものでしょうか？できればtags上のURLからもデータを取得したいのです｡

3. keyword.confにURLをそのまま貼り付けて実行できると楽そうです｡

4. キーワードを設定しDLbatを実行時にsampleファイルのように設定時例

**talents/t0128fe	
series/srookmi9gb**

から

**talents/t0128fe　	#ムロツヨシ
series/srookmi9gb	#かまいたちの知らんけど**

のように番組名などを取得し追記することは可能ですか？またその場合､URL貼り付け実行した場合の"ttps://tver.jp/"は削除していただくと従来のような見た目の整然さは保てそうです｡

5. list生成を実行するのに引用するkeyword.confを例えばlistkeyword.conf等の別のファイルにすることは可能ですか？

お手数ですが検討お願い致します｡

**Reply by dongaba** [reply]
1. 既にBOMなしUTFでリリースしてしまっているので、無用な混乱を避けるためにBOMなしを継続していますが、ご自分でBOM付きにエンコードし直していただければ、TVeRecは以降BOM付きで追記し続けます。
2. 件数が違うことはないような気がしますが、もし件数が違うのであれば、TVerの仕様です。もしかしたら、内部で使っているAPIが変更になったのかもしれませんね。なお、私の環境ではどちらも969件で一致しています。
3. そうですね。でも、URLじゃないものもありますので。。というわけで、今のところ対応予定はありません。
4. #以降は無視するようになっているので、追記していただいても問題ないと思います。もしかしたら#の前には空白やタブが必要かもしれません。(どういうロジックだったか忘れましたので、サンプルと同様に記載していただければ確実です)
5. 可能かと言われれば可能です。。今のところ実装予定ありませんので、別のフォルダにもう1つTVerRecを配置するなどしてご対応いただければと思います。

**Reply by ghost** [reply]
早速の回答ありがとうございます！
135について了解しましたありがとうございます！
2については試せていないので後に試します｡

4は手動記載したいという事ではなく､バッチファイルで引用されたあとの動作(またはそれ用の取得バッチ)として各行にseriiesなら番組名などを自動追記していただけるとありがたいという意図でした｡
ありがとうございました｡
